### PR TITLE
Serve modules via requirejs

### LINKED ISSUE #83: Redo client-side module eval
Rather than passing the whole module as a single string, have the clients fetch the code from the server.

### PR DESCRIPTION
**Pull request by applmak**
This replaces sending the code over our network channel, and instead sends over only the path to the module's main file. It opens up (though does not *quite* allow) the possibility for multi-file modules.

Fixes #83.

### CODE REVIEW

**Comment by applmak** [comment]
I included the service locator code so you could see what it looked like. I think it's just hugely intrusive in the module code for things like 'debug' which you just wanna be global and not pass around.

File: config/demo-playlist.json [hunk 1]
```diff
@@ -99,6 +99,23 @@
         }
       }
     },
+    {
```
Comment (iatkinson) [inline config/demo-playlist.json]: Is this part of this change?

Comment (applmak) [inline config/demo-playlist.json]: It's not necessary for this PR. I can remove it if you'd like the PR to be more pure, and send it along as its own PR instead...

Comment (iatkinson) [inline config/demo-playlist.json]: I think it would be better to have in a separate PR.

Comment (applmak) [inline config/demo-playlist.json]: Done!
